core/src/main/java/org/mobilitydata/gtfsvalidator/parsing/RowParser.java

```diff
@@ -216,6 +217,18 @@ public String asText(int columnIndex, boolean required) {
 
   @Nullable
   public String asId(int columnIndex, boolean required) {
+    String value = row.asString(columnIndex);
+    if (value == null) {
+      return asString(columnIndex, required);
+    }
+    for (char ch : value.toCharArray()) {
```

Comment:
This makes a copy of the string that is a small performance issue. It is better to avoid them from the beginning rather than fixing them later. Please use a usual cycle here:

for(int i = 0, n = value.length() ; i < n ; i++) { 
 ...

Reply:
Good piece of advice. Strings are full of hidden quirks.
🗨️ https://www.baeldung.com/java-string-performance 🗨️